### PR TITLE
Fix non-CUDA builds after Windows PRs

### PR DESCRIPTION
**Pull request by apaszke**

### CODE REVIEW

File: torch/lib/THPP/tensors/generic/THTensor.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #define const_storage_cast(storage) \
   dynamic_cast<const THStorage<real>&>(storage)
 #define const_long_cast(tensor) \
-  dynamic_cast<const THTensor<std::conditional<sizeof(long) == 8, long, int64_t>::type>&>(tensor)
```
Comment (apaszke) [inline torch/lib/THPP/tensors/generic/THTensor.cpp]: @peterjc123 what's the reason for adding these templates? they seem to break the build on OS X

Comment (peterjc123) [reply torch/lib/THPP/tensors/generic/THTensor.cpp]: Just to use int64_t cast for MSVC. The long type takes up 4 bytes using MSVC while 8 bytes using GCC. If it fails to build, the following macro could be used as a second choice.
```Cpp
#ifndef _WIN32
#define const_long_cast(tensor) \
   dynamic_cast<const THTensor<long>::type>&>(tensor)
#else
#define const_long_cast(tensor) \
   dynamic_cast<const THTensor<int64_t>::type>&>(tensor)
#endif
```

Comment (peterjc123) [reply torch/lib/THPP/tensors/generic/THTensor.cpp]: It was originally introduced for consistency. @fmassa don't know the difference between long long and long in THPP. So i just keep using long for Unix systems here.

File: torch/csrc/generic/serialization.cpp [hunk 2]
```diff
@@ -6,21 +6,21 @@
 
 void THPTensor_(writeMetadataRaw)(THTensor *self, int fd)
 {
-  SYSCHECK(write(fd, &self->nDimension, sizeof(long)));
-  SYSCHECK(write(fd, self->size, sizeof(long) * self->nDimension));
-  SYSCHECK(write(fd, self->stride, sizeof(long) * self->nDimension));
-  SYSCHECK(write(fd, &self->storageOffset, sizeof(long)));
+  SYSCHECK(write(fd, &self->nDimension, sizeof(self->nDimension)));
```
Comment (fmassa) [inline torch/csrc/generic/serialization.cpp]: Hum... Was this a bug previously? `nDimension` was always `int`, but we were reading/writing as `long`.
Won't this change make reading files backward-incompatible?

**Pull request merged**